core/src/main/scala/org/apache/spark/storage/StorageLevel.scala

```diff
@@ -163,6 +163,20 @@ object StorageLevel {
   val MEMORY_AND_DISK_SER_2 = new StorageLevel(true, true, false, false, 2)
   val OFF_HEAP = new StorageLevel(true, true, true, false, 1)
 
+  /**
+   * :: DeveloperApi ::
+   * Return the StorageLevel object with the specified description.
+   */
+  @DeveloperApi
+  def fromDescription(desc: String): StorageLevel = {
```

Comment:
I'm a bit worried that we're relying on string parse which is easy to be broken. Is this covered by UT or so?

Reply:
Good suggestion, I'll add a unit test for it.